src/main/java/org/threadly/db/aurora/AuroraDataSource.java

```diff
@@ -0,0 +1,321 @@
+package org.threadly.db.aurora;
+
+import java.io.PrintWriter;
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import javax.naming.NamingException;
+import javax.naming.Reference;
+import javax.naming.Referenceable;
+import javax.naming.StringRefAddr;
+import javax.sql.DataSource;
+
+public final class AuroraDataSource implements DataSource, Referenceable, Serializable {
+  
+  private static final String AURORA_PREFIX = "jdbc:mysql:aurora:";
+  
+  private static final long serialVersionUID = 1L;
+  
+  /** The driver to create connections with */
+  private static final NonRegisteringDriver driver;
+  
+  static {
+    driver = new NonRegisteringDriver();
+  }
+  
+  /** Log stream */
+  protected transient PrintWriter logWriter = null;
+  
+  /** Database Name */
+  protected String databaseName = null;
+  
+  /** Character Encoding */
+  protected String encoding = null;
+  
+  /** Hostname */
+  protected String hostName = null;
+  
+  /** Password */
+  protected String password = null;
+  
+  /** The profileSQL property */
+  protected String profileSQLString = "false";
+  
+  /** The JDBC URL */
+  protected String url = null;
+  
+  /** User name */
+  protected String user = null;
+  
+  /** Should we construct the URL, or has it been set explicitly */
+  protected boolean explicitUrl = false;
+  
+  /** Port number */
+  protected int port = 3306;
+  
+  /**
+   * Default no-arg constructor for Serialization
+   */
+  public AuroraDataSource() {
+    // Default constructor for Serialization...
+  }
+  
+  @Override
+  public Logger getParentLogger() throws SQLFeatureNotSupportedException {
+    return null;
```

Comment:
Does it make sense to delegate to the driver implementation for this?  I see this is what mysql is doing as well, but I don't see why based off the javadocs: https://docs.oracle.com/javase/7/docs/api/javax/sql/CommonDataSource.html#getParentLogger()

Same goes for the below functions as well.

Reply:
I don't think we have a parent logger here ...